crates/muvm/src/bin/muvm.rs

```diff
@@ -383,68 +362,68 @@ fn main() -> Result<ExitCode> {
+        uid: getuid().as_raw(),
+        gid: getgid().as_raw(),
+        host_display: display,
+        server_cookie: cookie,
+        merged_rootfs: options.merged_rootfs,
+    };
+    let mut muvm_config_file = NamedTempFile::new()
+        .context("Failed to create a temporary file to store the muvm guest config")?;
+    write!(
+        muvm_config_file,
+        "{}",
+        serde_json::to_string(&guest_config)
+            .context("Failed to transform GuestConfiguration into a JSON string")?
+    )
+    .context("Failed to write to temporary config file")?;
+
+    let muvm_config_path = muvm_config_file
+        .path()
+        .to_str()
+        .context("Temporary directory path contains invalid UTF-8")?
+        .to_owned();
+    let muvm_guest_args = vec![
         muvm_guest_path
             .to_str()
             .context("Failed to process `muvm-guest` path as it contains invalid UTF-8")?
             .to_owned(),
-        username,
-        format!("{uid}", uid = getuid().as_raw()),
-        format!("{gid}", gid = getgid().as_raw()),
-        muvm_server_path
-            .to_str()
-            .context("Failed to process `muvm-server` path as it contains invalid UTF-8")?
-            .to_owned(),
-        command
-            .to_str()
-            .context("Failed to process command as it contains invalid UTF-8")?
-            .to_string(),
+        muvm_config_path,
     ];
-    for arg in command_args {
-        muvm_guest_args.push(arg);
-    }
 
-    env.insert(
-        "MUVM_SERVER_PORT".to_owned(),
-        options.server_port.to_string(),
-    );
-    env.insert(
-        "MUVM_ROOT_SERVER_PORT".to_owned(),
-        options.root_server_port.to_string(),
-    );
-    env.insert("MUVM_SERVER_COOKIE".to_owned(), cookie.to_string());
-
-    if !options.sommelier {
-        let display =
-            env::var("DISPLAY").context("X11 forwarding requested but DISPLAY is unset")?;
-        env.insert("HOST_DISPLAY".to_string(), display);
-
-        // And forward XAUTHORITY. This will be modified to fix the
-        // display name in muvm-guest.
-        if let Ok(xauthority) = env::var("XAUTHORITY") {
-            env.insert("XAUTHORITY".to_string(), xauthority);
-        }
+    // And forward XAUTHORITY. This will be modified to fix the
+    // display name in muvm-guest.
+    if let Ok(xauthority) = env::var("XAUTHORITY") {
+        env.insert("XAUTHORITY".to_owned(), xauthority);
     }
 
-    let mut krun_config = KrunConfig {
-        args: Vec::new(),
-        envs: Vec::new(),
-    };
-    for arg in muvm_guest_args {
-        krun_config.args.push(arg);
-    }
-    for (key, value) in env {
-        krun_config.envs.push(format!("{}={}", key, value));
-    }
     let krun_config = KrunBaseConfig {
-        config: krun_config,
+        config: KrunConfig {
+            args: muvm_guest_args,
+            envs: env
+                .into_iter()
+                .map(|(key, value)| format!("{}={}", key, value))
```

Comment:
```suggestion
                .map(|(key, value)| format!("{key}={value}"))
```

No reason not to use the more readable syntax, since our MSRV is 1.77 >= 1.58